style50/style50.py

```diff
@@ -179,13 +182,17 @@ def _check(self, file):
         _, extension = os.path.splitext(file)
         try:
             check = self.extension_map[extension[1:]]
-
-            with open(file) as f:
-                code = f.read()
         except KeyError:
-            raise Error("unknown file type \"{}\", skipping...".format(file))
-        else:
-            return check(code)
+            magic_type = magic.from_file(file)
+            for names, cls in self.magic_map.items():
+                if any(name in magic_type for name in names):
```

Comment:
Does `any` short-circuit these iterations? If not, is intersection cleaner? Just curious.

Reply:
Yes, any short circuits.